camel/src/main/java/com/github/theprez/manzan/WatchStarter.java

```diff
@@ -0,0 +1,123 @@
+package com.github.theprez.manzan;
+
+import java.beans.PropertyVetoException;
+import java.io.IOException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.ini4j.InvalidFileFormatException;
+
+import com.github.theprez.manzan.configuration.ApplicationConfig;
+import com.ibm.as400.access.AS400;
+import com.ibm.as400.access.AS400Bin4;
+import com.ibm.as400.access.AS400Message;
+import com.ibm.as400.access.AS400SecurityException;
+import com.ibm.as400.access.AS400Text;
+import com.ibm.as400.access.CommandCall;
+import com.ibm.as400.access.ErrorCodeParameter;
+import com.ibm.as400.access.ErrorCompletingRequestException;
+import com.ibm.as400.access.ObjectDoesNotExistException;
+import com.ibm.as400.access.ProgramCall;
+import com.ibm.as400.access.ProgramParameter;
+
+public class WatchStarter {
+    private final String m_command;
+    private final String m_stopCommand;
+    private final String m_session_id;
+
+    public WatchStarter(final String _session_id, final String _deets) throws InvalidFileFormatException, IOException {
+
+        String command = _deets;
+        String wchpgm = " WCHPGM(" + ApplicationConfig.get().getLibrary() + "/HANDLER) ";
+        command = command.replaceAll("(?i)(^|\\s+)wchpgm\\([A-Z0-9\\/\\*\\s]+\\)\\s*", " ");
+
+        String ssnid = " SSNID(" + _session_id + ") ";
+        command = command.replaceAll("(?i)(^|\\s+)ssnid\\([A-Z0-9\\*\\s]+\\)\\s*", " ");
+
+        command = command.replaceFirst("^(?i)([a-z0-9]+\\/)?strwch\\s+", " ");
+        command = "QSYS/STRWCH" + ssnid + wchpgm + command;
+        m_command = command;
+        m_stopCommand = "QSYS/ENDWCH SSNID(" + _session_id.trim() + ")";
+        m_session_id = _session_id;
+    }
+
+    public void endwch()
+            throws AS400SecurityException, ErrorCompletingRequestException, IOException, InterruptedException {
+        System.out.println("ending watch");
+        runCmd(m_stopCommand);
+    }
+
+    public void strwch()
+            throws IOException, AS400SecurityException, ErrorCompletingRequestException, InterruptedException,
+            PropertyVetoException, ObjectDoesNotExistException {
+        if (isRunning()) {
+            System.err.println("Watch '" + m_session_id + "' already running");
+            return;
+        }
+        runCmd(m_command);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Will this be invoked even if the process is killed, or if interrupted with Control+C?

Reply:
> Will this be invoked even if the process is killed, or if interrupted with Control+C? 

Yep. ENDJOb too. 

It won't run in cases of immediate job/vm termination, though, like:
- stopping from the debugger
- JVM crash

Reply:
I am ok with that. Thanks!